ui/src/buckets/actions/index.ts

```diff
@@ -105,12 +109,13 @@ export const createBucket = (bucket: Bucket) => async (
       orgs: {org},
     } = getState()
 
-    const createdBucket = await client.buckets.create({
-      ...bucket,
-      orgID: org.id,
-    })
+    const resp = await api.postBucket({data: {...bucket, orgID: org.id}})
 
-    dispatch(addBucket(createdBucket))
+    if (resp.status !== 201) {
+      throw new Error(resp.data.message)
+    }
+
+    dispatch(addBucket(resp.data))
     dispatch(checkBucketLimits())
```

Comment:
I think this line (and other checking of asset limits) needs to run both in the case of resp.status=200 and otherwise. can we move them above the error?

Reply:
It looks like the only possible responses are `201` and `500`:

```
    post:
      operationId: PostBuckets
      tags:
        - Buckets
      summary: Create a bucket
      parameters:
          - $ref: '#/components/parameters/TraceSpan'
      requestBody:
        description: bucket to create
        required: true
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/Bucket"
      responses:
        '201':
          description: Bucket created
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Bucket"
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
```

Reply:
yes and we should check bucket limits in both cases.

Reply:
ohh I see. So if it returns a `500`, that's because it may have hit the rate limit? And `dispatch(checkBucketLimits())` would be what eventually shows an error to the user?

Reply:
I guess I'm a little fuzzy on why we need to check the rate limits for buckets if we tried to add a bucket but failed. I was thinking we'd have the same exact number of buckets in that case, so checking the rate limit would be unnecessary?

Reply:
The requirements were to display a static error banner on top of buckets list- rather than just a notification- so yes if you're at the last bucket you can create you succeed in creation and then check your limits. I apparently designed it to only check limits in case of create rather than on get, so I was relying on create to display the warning. I'll add a checkLimits on get to make it work in either case.

Reply:
ah, that makes sense. Ok, thank you!